diff --git a/src/shared/clrdefinitions.cmake b/src/shared/clrdefinitions.cmake
index 0a7a73b..697e29d 100644
--- a/src/shared/clrdefinitions.cmake
+++ b/src/shared/clrdefinitions.cmake
@@ -101,7 +101,7 @@ if(FEATURE_EVENT_TRACE)
     add_compile_definitions(FEATURE_EVENT_TRACE)
     add_definitions(-DFEATURE_PERFTRACING)
 else(FEATURE_EVENT_TRACE)
-    add_custom_target(eventing_headers) # add a dummy target to avoid checking for FEATURE_EVENT_TRACE in multiple places
+#    add_custom_target(eventing_headers) # add a dummy target to avoid checking for FEATURE_EVENT_TRACE in multiple places
 endif(FEATURE_EVENT_TRACE)
 if(FEATURE_GDBJIT)
     add_definitions(-DFEATURE_GDBJIT)
