diff --git a/src/shared/pal/src/configure.cmake b/src/shared/pal/src/configure.cmake
index dd0e26e..0a45864 100644
--- a/src/shared/pal/src/configure.cmake
+++ b/src/shared/pal/src/configure.cmake
@@ -95,6 +95,9 @@ if (HAVE_LIBPTHREAD)
   set(PTHREAD_LIBRARY pthread)
 elseif (HAVE_PTHREAD_IN_LIBC)
   set(PTHREAD_LIBRARY c)
+else()
+  # On macOS and other systems, pthreads may be built into the system library
+  set(PTHREAD_LIBRARY c)
 endif()
 
 check_library_exists(${PTHREAD_LIBRARY} pthread_attr_get_np "" HAVE_PTHREAD_ATTR_GET_NP)
@@ -124,7 +127,11 @@ check_function_exists(directio HAVE_DIRECTIO)
 check_function_exists(semget HAS_SYSV_SEMAPHORES)
 check_function_exists(pthread_mutex_init HAS_PTHREAD_MUTEXES)
 check_function_exists(ttrace HAVE_TTRACE)
-check_function_exists(pipe2 HAVE_PIPE2)
+if(APPLE)
+    set(HAVE_PIPE2 FALSE CACHE INTERNAL "pipe2 is missing on macOS")
+else()
+    check_function_exists(pipe2 HAVE_PIPE2)
+endif()
 check_function_exists(strerrorname_np HAVE_STRERRORNAME_NP)
 
 check_cxx_source_compiles("
